modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py

```diff
@@ -508,3 +510,10 @@ def _validate_read_csv_kwargs(
 
         if on_bad_lines not in ["error", "warn", "skip", None]:
             raise ValueError(f"Argument {on_bad_lines} is invalid for on_bad_lines.")
+
+    @classmethod
+    @_inherit_docstrings(BaseIO.read_sql, apilink="pandas.read_sql")
+    def read_sql(cls, **kwargs):
```

Comment:
Why override the base implementation?

Reply:
To remove the arguments, not declared in the base implementation. Otherwise, some tests fail:
```
E       TypeError: read_sql() got an unexpected keyword argument 'partition_column'
```

Reply:
Let's make it more explicit that `HDK` doesn't support `partition_column`, `lower_bound`, `upper_bound`, `max_sessions` options, instead of just ignoring them. If the specified options are other than `None`, then we need to throw an `NotImplementedError` exception. If they are `None`, then just remove them from `kwargs`.

Reply:
Does it make sense to add this logic to the base implementation?